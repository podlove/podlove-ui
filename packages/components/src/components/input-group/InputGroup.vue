<script lang="ts" setup>
import { ref, onMounted } from 'vue';

const elements = ref(0);
const container = ref<HTMLElement | null>(null);

onMounted(() => {
  console.log(container.value)
  if (!container.value) {
    return;
  }

  elements.value = container.value.children.length;
})
</script>

<template>
  <div class="podlove-component-input-group h-8 box-border flex" :class="{ [`elements-${elements}`]: true }" ref="container">
    <slot name="button" />
    <slot name="input" />
  </div>
</template>

<style lang="scss">
$addon-button-width: 80px;

.podlove-component-input-group {
  .input-text,
  .input-select {
    padding: 0.25rem;
    width: 100%;
    border-radius: 0 3px 3px 0;
    border-width: 1px 1px 1px 0;
  }

  .button {
    width: $addon-button-width;
    display: block;
    border-width: 1px;
    border-radius: 3px 0 0 3px;
  }

  &.elements-2 {
    *:nth-child(1) {
      width: auto;
      border-radius: 3px 0 0 3px;
    }
  }

  &.elements-3 {
    *:nth-child(1) {
      width: auto;
      border-radius: 3px 0 0 3px;
    }

    *:nth-child(2) {
      width: auto;
      border-radius: 0;
    }
  }
}
</style>
