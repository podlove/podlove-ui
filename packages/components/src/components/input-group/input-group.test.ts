import { describe, test, expect } from 'vitest';
import { mount } from '@vue/test-utils';

import InputGroup from './InputGroup.vue';

describe('InputGroup', () => {
  describe('component', () => {
    test('should be a Vue instance', async () => {
      const wrapper = await mount(InputGroup);
      expect(wrapper.vm).toBeTruthy();
    });
  });

  describe('slots', () => {
    test('should have a slot for inputs', async () => {
      const wrapper = await mount(InputGroup, {
        slots: {
          input: '<input type="text" />'
        }
      });

      expect(wrapper.element).toMatchSnapshot();
    });

    test('should have a slot for buttons', async () => {
      const wrapper = await mount(InputGroup, {
        slots: {
          button: '<button>click!</button>'
        }
      });

      expect(wrapper.element).toMatchSnapshot();
    });

    test('should apply special classes when having multiple slots', async () => {
      const wrapper = await mount(InputGroup, {
        slots: {
          button: '<button>click!</button>',
          input: ['<input type="text" />', '<input type="text" />']
        }
      });

      expect(wrapper.element).toMatchSnapshot();
    });
  });
});
