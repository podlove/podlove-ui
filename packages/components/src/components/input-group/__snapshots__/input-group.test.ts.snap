// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`InputGroup > slots > should apply special classes when having multiple slots 1`] = `
<div
  class="podlove-component-input-group h-8 box-border flex elements-6"
>
  
  <button>
    click!
  </button>
  
  
  
  <input
    type="text"
  />
  
  
  <input
    type="text"
  />
  
  
</div>
`;

exports[`InputGroup > slots > should have a slot for buttons 1`] = `
<div
  class="podlove-component-input-group h-8 box-border flex elements-1"
>
  
  <button>
    click!
  </button>
  
  
  
</div>
`;

exports[`InputGroup > slots > should have a slot for inputs 1`] = `
<div
  class="podlove-component-input-group h-8 box-border flex elements-1"
>
  
  
  
  <input
    type="text"
  />
  
</div>
`;
